howtos/how_to_request_cached_data.py

```diff
@@ -0,0 +1,139 @@
+                                                                                           cache_access_timeout)
+        correlation_id = 12345
+        correlation_completion_listener = MyCorrelationCompletionListener()
+
+        direct_message_receiver.requestCached(cached_message_subscription_request, correlation_id, correlation_completion_listener)
+
+    def create_subscription_request_to_receive_latest_message(direct_message_receiver: 'DirectMessageReceiver',
+                                                              subscription_expression: str,
+                                                              cache_name: str,
+                                                              cache_access_timeout: int):
+        """
+        An example of how to create and perform CachedMessageSubscriptionRequest to receive latest messages. When no live
+        messages are available, cached messages matching specified topic subscription considered latest, live messages otherwise.
+        When live messages are available then cached messages are discarded.
+
+        Args:
+            direct_message_receiver(DirectMessageReceiver): An already configured and connected receiver.
+            subscription_expression(str): Topic subscription expression.
+            cache_name(str): name of the solace cache to retrieve from
+            cache_access_timeout(int): Solace cache request timeout, in milliseconds
+
+        Raises:
+            InterruptedException: If any thread has interrupted the current thread
+            PubSubPlusClientException: If the operation could not be performed
+            IllegalStateException: If the service is not running
+        """
+        cached_message_subscription_request_for_latest_of_cached_or_live_messages = \
+            CachedMessageSubscriptionRequest.live_cancels_cached(cache_name,
+                                                                 TopicSubscription.of(subscriptionExpression),
+                                                                 cache_access_timeout)
+
+        correlation_id = 12345
+        correlation_completion_listener = MyCorrelationCompletionListener()
+
+        direct_receiver.request_cached(cached_message_subscription_request_for_latest_of_cached_or_live_messages,
+                                       correlation_id,
+                                       correlation_completion_listener)
+
+    def create_cached_subscription_with_more_options(direct_message_receiver: 'DirectMessageReceiver',
+                                                     subscription_expression: 'SubscriptionExpression',
+                                                     cache_name: str,
+                                                     cache_access_timeout: int,
+                                                     max_cached_messages: int,
```

Comment:
Is there a reason we are showing the user `max_cached_messages`/`cached_message_age` in this example and none of the others? They all have this optional argument available.

Reply:
Those parameters are included to show that the options can be passed, in contrast to the other examples where they are not passed. Would it be better if this were communicated differently?

Reply:
I'm aware `max_cached_messages` and `cached_message_age` can be passed in for all the cache request types but shouldn't we then show that for all the samples? Only showing it on this one give the impression that there is something special about this method.

Reply:
What do you think would be a better way of describing that the parameters are optional? Do you think it would be more clear to have the parameters in every function, and just say in the doc string that they are optional?

Reply:
> What do you think would be a better way of describing that the parameters are optional

Including the word optional somewhere in the parameter description would be good

> Do you think it would be more clear to have the parameters in every function

If there is a specific reason it was added on just this one it's okay, it should just be called out as to why.  If there was no specific reason for just this method then yes they should be shown on all the methods

Reply:
I just noticed that the method name is `create_cached_subscription_with_more_options`. Do you think that the method name is indicative enough that this sample is intended to show how to configure a CachedMessageSubscription with more options than the defaults?

Reply:
Yeah I think that's okay